src/libraries/System.Private.CoreLib/src/System/TimeOnly.cs

```diff
@@ -84,68 +84,66 @@ public TimeOnly(long ticks)
                 throw new ArgumentOutOfRangeException(nameof(ticks), SR.ArgumentOutOfRange_TimeOnlyBadTicks);
             }
 
-            _ticks = ticks;
+            _ticks = (ulong)ticks;
         }
 
         // exist to bypass the check in the public constructor.
-        internal TimeOnly(ulong ticks) => _ticks = (long)ticks;
+        internal TimeOnly(ulong ticks) => _ticks = ticks;
 
         /// <summary>
         /// Gets the hour component of the time represented by this instance.
         /// </summary>
-        public int Hour => new TimeSpan(_ticks).Hours;
+        public int Hour => (int)(_ticks / TimeSpan.TicksPerHour);
 
         /// <summary>
         /// Gets the minute component of the time represented by this instance.
         /// </summary>
-        public int Minute => new TimeSpan(_ticks).Minutes;
+        public int Minute => (int)((uint)(_ticks / TimeSpan.TicksPerMinute) % 60);
```

Comment:
>60 [](http://example.com/codeflow?start=78&length=2)

Will be good to use the constants instead (like MinutesPerHour).

Reply:
Changed, but I'm not sure if it actually improves readability for such basic constants...